src/Function.cpp

```diff
@@ -248,6 +338,103 @@ Function::Function(const std::string &n) : contents(new FunctionContents) {
     contents.ptr->name = n;
 }
 
+void deep_copy_update_definition_helper(IntrusivePtr<FunctionContents> &contents,
+                                        UpdateDefinition &dst,
+                                        const UpdateDefinition &src,
+                                        DeepCopyMap &copied_map) {
+    dst.values = src.values;
+    dst.args = src.args;
+    dst.schedule = Schedule();
+    src.schedule.deep_copy(dst.schedule, copied_map);
```

Comment:
Another option which I've seen elsewhere in this branch is to make a Schedule::deep_copy(DeepCopyMap &) member function of Schedule that returns a deep copy. That would be consistent with e.g. ReductionDomain::deep_copy.